### PR TITLE
issue#33: devon4j architecture tutorial

### PR DESCRIPTION
**Pull request by ssarmokadam**

### CODE REVIEW

File: devon4j-architecture/principles.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+*  [Information Hiding](http://en.wikipedia.org/wiki/Information_hiding) via component API and its exchangeable implementation treated as secret.
+*  _Design by Contract_ for self-contained, descriptive, and stable component APIs.
+*  [Layering](https://github.com/devonfw/devon4j/blob/master/documentation/architecture.asciidoc#technical-architecture) as well as separation of business logic from technical code for better maintenance.
+*  _Data Sovereignty_ (and _high cohesion with low coupling_) says that a component is responsible for its data and changes to this data shall only happen via the component. Otherwise, maintenance problems will arise to ensure that data remains consistent. Therefore, interfaces of a component that may be used by other components are designed _call-by-value_ and not _call-by-reference_.
+* *Homogeneity* 
+Solve similar problems in similar ways and establish a uniform [code-style](https://github.com/devonfw/devon4j/blob/master/documentation/coding-conventions.asciidoc).
+
+As an architect you should be prepared for the future by reading the [TechnoVision](https://www.capgemini.com/de-de/wp-content/uploads/sites/5/2020/07/TechnoVision-2020-Report.pdf).
```
Comment (SchettlerKoehler) [inline devon4j-architecture/principles.md:29]: @maybeec do you think we should have this link here?

File: devon4j-architecture/index.asciidoc [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+--
+restoreDevonfwIde(["java", "mvn"])
+--
+Before proceeding to next steps ide should be installed.
+====
+====
+In this, we will create sample devon4j application with name sampleapp. 
+[step]
+--
+createDevon4jProject("sampleapp")
```
Comment (SchettlerKoehler) [inline devon4j-architecture/index.asciidoc:37]: I think you are creating the project to show the user the architecture, but this is missing. You should point the user to the parts of the project where the things you are mentioned are visible.

**Review by SchettlerKoehler — CHANGES_REQUESTED** [review]

**Review by SchettlerKoehler — CHANGES_REQUESTED** [review]

File: devon4j-architecture/index.asciidoc [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+====
+### Step1 : Install devonfw-ide
+
+To create devon4j application it is necessary to have devonfw-ide installed on your machine.For this tutorial purpose we will have it installed here.
+
+If you want to try it on local machine you can refer [here](https://devonfw.com/website/pages/docs/devonfw-ide-introduction.asciidoc.html#setup.asciidoc).
+
+[step]
+--
+installDevonfwIde(["java","mvn","vscode"])
```
[the remark this answers is not in the document]
Comment (ssarmokadam) [reply devon4j-architecture/index.asciidoc]: Done

File: devon4j-architecture/index.asciidoc [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+Before proceeding to next steps make sure ide is installed.
+====
+====
+### Step2: Create devon4j sample application
+
+Once devonfw-ide is installed , we will create sample devon4j application with name *sampleapp*. 
+
+If you want to create devon4j application on your local there are two ways to do it:
+* In eclipse as shown [here](https://devonfw.com/website/pages/docs/devon4j.asciidoc_tutorials.html#tutorial-newapp.asciidoc_from-eclipse)
+* Using command line as shown [here](https://devonfw.com/website/pages/docs/devon4j.asciidoc_tutorials.html#tutorial-newapp.asciidoc_from-command-line)
```
[the remark this answers is not in the document]
Comment (ssarmokadam) [reply devon4j-architecture/index.asciidoc:26]: I have rearranged text to make it clear

File: devon4j-architecture/index.asciidoc [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+createDevon4jProject("sampleapp")
+--
+
+Once project is created switch to next tab of IDE. In IDE explorer you can see folder structure like devonfw-> workspaces->main->sampleapp . In sampleapp modules such as api,core and server will get created.
+
+====
+
+[step]
+--
+nextKatacodaStep("devon4j application architecture", [{ "image": "images/devon4j_proj_structure1.jpg" }])
```
[the remark this answers is not in the document]
Comment (ssarmokadam) [reply devon4j-architecture/index.asciidoc:42]: Done

[the remark this answers is not in the document]
Comment (ssarmokadam) [reply devon4j-architecture/index.asciidoc:42]: I have added links of wiki as we do not want to repeat it

**Comment by MarcelDiessner** [comment]
You can now add a conclusion text which is shown at the end of the katacoda tutorial and the wiki. See https://github.com/devonfw-tutorials/tutorials/wiki/Development for further informations

**Comment by MarcelDiessner** [comment]
You can now add a subtitle with the == syntax. See https://github.com/devonfw-tutorials/tutorials/wiki/Development

**Review by sujith-mn — APPROVED** [review]

**Pull request merged**